backend/src/apiserver/config/driver_config.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2025 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+const (
+	// Environment variable names for driver pod configuration
+	EnvDriverPodLabels      = "DRIVER_POD_LABELS"
+	EnvDriverPodAnnotations = "DRIVER_POD_ANNOTATIONS"
+
+	// Reserved label prefix that should be filtered out
+	ReservedLabelPrefix = "pipelines.kubeflow.org/"
+)
+
+// DriverPodConfig holds the configuration for driver pod labels and annotations
+type DriverPodConfig struct {
+	Labels      map[string]string `json:"labels"`
+	Annotations map[string]string `json:"annotations"`
+}
+
+// GetDriverPodConfig reads driver pod configuration from environment variables.
+// It supports both JSON format and comma-separated key=value format.
+// Reserved labels with prefix "pipelines.kubeflow.org/" are filtered out.
+// Returns an empty config (not nil) on errors to allow graceful degradation.
+func GetDriverPodConfig() (*DriverPodConfig, error) {
+	config := &DriverPodConfig{
+		Labels:      make(map[string]string),
+		Annotations: make(map[string]string),
+	}
+
+	// Read and parse labels from environment variable
+	labelsEnv := os.Getenv(EnvDriverPodLabels)
+	if labelsEnv != "" {
+		labels, err := parseConfigValue(labelsEnv)
+		if err != nil {
+			glog.Warningf("Failed to parse %s: %v. Using empty labels.", EnvDriverPodLabels, err)
+		} else {
+			config.Labels = labels
+		}
+	}
+
+	// Read and parse annotations from environment variable
+	annotationsEnv := os.Getenv(EnvDriverPodAnnotations)
+	if annotationsEnv != "" {
+		annotations, err := parseConfigValue(annotationsEnv)
+		if err != nil {
+			glog.Warningf("Failed to parse %s: %v. Using empty annotations.", EnvDriverPodAnnotations, err)
+		} else {
+			config.Annotations = annotations
+		}
+	}
+
+	// Filter out reserved system labels
+	validateSystemLabels(config)
+
+	return config, nil
+}
+
+// parseConfigValue attempts to parse the input as JSON first,
+// then falls back to parsing as comma-separated key=value pairs
+func parseConfigValue(input string) (map[string]string, error) {
+	input = strings.TrimSpace(input)
+	if input == "" {
+		return make(map[string]string), nil
+	}
+
+	// Try parsing as JSON first
+	if strings.HasPrefix(input, "{") {
```

Comment:
Let's just stick to the JSON format since that aligns better with `config.json` configuration format and it keeps it consistent.

Reply:
Also, let's assume the format is the same as labels and annotations in a Kubernetes object.